### PR TITLE
Add statement LOOP AT SCREEN to parser

### LINKED ISSUE #314: LOOP AT SCREEN not possible in cloud
update syntax diagram?

### PR DESCRIPTION
**Pull request by jan-jezek**
Closes #314

### CODE REVIEW

**Comment by jan-jezek** [comment]
Currently the parser takes LOOP AT SCREEN as loop statement. I don't know how to make it to not parse as loop, but as loopAtScreen @larshp?

File: packages/core/src/abap/2_statements/statements/index.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export * from "./endprovide";
 export * from "./if";
 export * from "./convert_text";
 export * from "./log_point";
+export * from "./loop_at_screen";
```
Comment (larshp) [inline packages/core/src/abap/2_statements/statements/index.ts:29]: try moving this up before line 23

Comment (jan-jezek) [reply packages/core/src/abap/2_statements/statements/index.ts]: That worked for this problem.

**Comment by jan-jezek** [comment]
The problem is that it is working the other way then I expected:
```
  it("LOOP AT SCREEN, on 702", () => {
    const abap = `LOOP AT SCREEN.
    ENDLOOP.`;
    const issues = runProgram(abap, [], Version.v702);
    expect(issues.length).to.equals(0);
  });

  it("LOOP AT SCREEN, on Cloud", () => {
    const abap = `LOOP AT SCREEN.
    ENDLOOP.`;
    const issues = runProgram(abap, [], Version.Cloud);
    expect(issues.length).to.equals(0);
  });
```

**Comment by larshp** [comment]
what is the error message for the first 702 testcase? I guess its a structure error, ie. it does not know yet that LOOP AT SCREEN and ENDLOOP must match, only normal LOOP and ENDLOOP works together. Need another one of https://syntax.abaplint.org/?filter=loop#/abap/structure/Loop

The statement https://syntax.abaplint.org/?filter=loop#/abap/statement/Loop probably still matches "LOOP AT SCREEN", need some way to not allow for a variable called "screen"

**Comment by jan-jezek** [comment]
I have added the new structure, but I don't know to make Loop not recognize LoopAtScreen...

**Comment by larshp** [comment]
hmm, yea, also not sure

**Comment by larshp** [comment]
working on this now...

**Comment by larshp** [comment]
passing locally, few notes
* used statementVersionFail to check the parser fails on a specific version for a statement
* new `fail()` added to the parser combinator

will merge if the regression test passes

**Comment by larshp** [comment]
🎉

**Comment by larshp** [comment]
will be published in 2.85.42

**Pull request merged**